crates/rspack_binding_values/src/path_data.rs

```diff
@@ -10,6 +12,47 @@ pub struct PathData {
   pub runtime: Option<String>,
   pub url: Option<String>,
   pub id: Option<String>,
+  pub chunk: Option<ChunkPathData>,
+}
+
+impl TryFrom<rspack_core::PathData<'_>> for PathData {
+  type Error = napi::Error;
+  fn try_from(path_data: rspack_core::PathData<'_>) -> Result<Self, Self::Error> {
+    Ok(Self {
+      filename: path_data.filename.map(|s| s.to_string()),
+      hash: path_data.hash.map(|s| s.to_string()),
+      content_hash: path_data.content_hash.map(|s| s.to_string()),
+      runtime: path_data.runtime.map(|s| s.to_string()),
+      url: path_data.url.map(|s| s.to_string()),
+      id: path_data.id.map(|s| s.to_string()),
+      chunk: path_data.chunk.map(ChunkPathData::from),
+    })
+  }
+}
+
+#[napi(object)]
+pub struct ChunkPathData {
+  pub id: Option<String>,
+  pub name: Option<String>,
+  pub hash: Option<String>,
+  pub content_hash: Option<HashMap<String, String>>,
+}
```

Comment:
Name consistent with the one in webpack's dts `https://github.com/webpack/webpack/blob/e4fee1527b5444da931be6694227143f1cba53c4/types.d.ts#L1423`.

More fields to be added. Existing ones are enough for most use.

We can't reuse https://github.com/web-infra-dev/rspack/blob/9c25b4d51e36d8f8a734874318d460201ae38695/crates/rspack_binding_values/src/chunk.rs#L9 because it has non-optional `inner*` fields, which would reject user-supplied `PathData` at https://github.com/web-infra-dev/rspack/blob/9c25b4d51e36d8f8a734874318d460201ae38695/crates/rspack_binding_values/src/compilation.rs#L378

Reply:
```suggestion
pub struct JsChunkPathData {
  pub id: Option<String>,
  pub name: Option<String>,
  pub hash: Option<String>,
  pub content_hash: Option<HashMap<String, String>>,
}
```

Prefixed with `Js` to make things more clear.

Reply:
Resolved in 0baf67455a1ed6dc2ab2f302024d093a18be7af5